sonar-scanner-engine/src/main/java/org/sonar/batch/postjob/DefaultPostJobContext.java

```diff
@@ -53,11 +56,17 @@ public Settings settings() {
 
   @Override
   public Iterable<PostJobIssue> issues() {
+    if (!analysisMode.isIssues()) {
```

Comment:
if you decide to throw an IllegalStateException, then the code can be simplified to:
```
Preconditions.checkState(analysisMode.isIssues(), "Issues are only available to PostJobs in 'issues' mode");
```
:-)